### PR TITLE
sensor calibrations (accel/gyro/mag) add calibration count

### PR DESCRIPTION
**Pull request by dagar**
- monotonically increasing count used to indicate if the sensor's calibration has changed

Later this can be used to reset the estimator rather than rebooting. 

https://github.com/PX4/Firmware/issues/9030

### CODE REVIEW

**Comment by dagar** [comment]
``` Console

nsh> listener vehicle_imu

TOPIC: vehicle_imu 3 instances

Instance 0:
 vehicle_imu_s
        timestamp: 6951079  (0.001176 seconds ago)
        timestamp_sample: 6950642  (437 us before timestamp)
        accel_device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
        gyro_device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
        delta_angle: [-0.0000, -0.0000, -0.0000]
        delta_velocity: [-0.0014, -0.0026, -0.0496]
        delta_angle_dt: 5062
        delta_velocity_dt: 5062
        delta_velocity_clipping: 0
        calibration_count: 1

Instance 1:
 vehicle_imu_s
        timestamp: 6953216  (0.000423 seconds ago)
        timestamp_sample: 6952959  (257 us before timestamp)
        accel_device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
        gyro_device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
        delta_angle: [-0.0000, 0.0000, -0.0000]
        delta_velocity: [-0.0005, -0.0031, -0.0495]
        delta_angle_dt: 5067
        delta_velocity_dt: 5067
        delta_velocity_clipping: 0
        calibration_count: 1

Instance 2:
 vehicle_imu_s
        timestamp: 6954021  (0.001281 seconds ago)
        timestamp_sample: 6953856  (165 us before timestamp)
        accel_device_id: 4259850 (Type: 0x41, SPI:1 (0x00))
        gyro_device_id: 4325386 (Type: 0x42, SPI:1 (0x00))
        delta_angle: [-0.0000, -0.0000, -0.0000]
        delta_velocity: [-0.0005, -0.0023, -0.0489]
        delta_angle_dt: 5013
        delta_velocity_dt: 5014
        delta_velocity_clipping: 0
        calibration_count: 1


nsh> commander calibrate gyro
nsh> INFO  [commander] [cal] calibration started: 2 gyro
INFO  [commander] [cal] progress <9>
INFO  [commander] [cal] progress <14>
INFO  [commander] [cal] progress <19>
INFO  [commander] [cal] progress <38>
INFO  [commander] [cal] progress <43>
INFO  [commander] [cal] progress <48>
INFO  [commander] [cal] progress <52>
INFO  [commander] [cal] progress <57>
INFO  [commander] [cal] progress <62>
INFO  [commander] [cal] progress <67>
INFO  [commander] [cal] progress <72>
INFO  [commander] [cal] progress <76>
INFO  [commander] [cal] progress <81>
INFO  [commander] [cal] progress <86>
INFO  [commander] [cal] progress <91>
INFO  [commander] [cal] progress <96>
INFO  [sensor_calibration] GYRO 3670026 EN: 1, offset: [0.0111 -0.0093 0.0061]
INFO  [sensor_calibration] GYRO 3932170 EN: 1, offset: [0.0031 0.0115 -0.0020]
INFO  [sensor_calibration] GYRO 4325386 EN: 1, offset: [0.0019 0.0049 0.0021]
INFO  [commander] [cal] calibration done: gyro


nsh> listener vehicle_imu

TOPIC: vehicle_imu 3 instances

Instance 0:
 vehicle_imu_s
        timestamp: 161707237  (0.000490 seconds ago)
        timestamp_sample: 161706902  (335 us before timestamp)
        accel_device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
        gyro_device_id: 3670026 (Type: 0x38, SPI:1 (0x00))
        delta_angle: [-0.0000, -0.0000, 0.0000]
        delta_velocity: [-0.0014, -0.0026, -0.0490]
        delta_angle_dt: 5002
        delta_velocity_dt: 5002
        delta_velocity_clipping: 0
        calibration_count: 1

Instance 1:
 vehicle_imu_s
        timestamp: 161709153  (0.000649 seconds ago)
        timestamp_sample: 161708901  (252 us before timestamp)
        accel_device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
        gyro_device_id: 3932170 (Type: 0x3C, SPI:1 (0x00))
        delta_angle: [0.0000, 0.0000, -0.0000]
        delta_velocity: [-0.0007, -0.0030, -0.0488]
        delta_angle_dt: 5004
        delta_velocity_dt: 5004
        delta_velocity_clipping: 0
        calibration_count: 2

Instance 2:
 vehicle_imu_s
        timestamp: 161711335  (0.000109 seconds ago)
        timestamp_sample: 161711217  (118 us before timestamp)
        accel_device_id: 4259850 (Type: 0x41, SPI:1 (0x00))
        gyro_device_id: 4325386 (Type: 0x42, SPI:1 (0x00))
        delta_angle: [-0.0000, -0.0000, -0.0000]
        delta_velocity: [-0.0007, -0.0023, -0.0482]
        delta_angle_dt: 5011
        delta_velocity_dt: 4961
        delta_velocity_clipping: 0
        calibration_count: 2
```

**Comment by dagar** [comment]
> Later this can be used to reset the estimator rather than rebooting.

In addition to that I'm going to use this mechanism to save the EKF2 learned mag biases back to the existing per sensor instance calibration parameters rather than separately stored directly in EKF2.

**Comment by dagar** [comment]
This mechanism is what I'll be using to determine when to broadcast Mavlink `MAG_CAL_REPORT` (https://mavlink.io/en/messages/common.html#MAG_CAL_REPORT) on calibration change.

File: src/lib/sensor_calibration/Accelerometer.cpp [hunk 1]
```diff
@@ -149,11 +149,27 @@ void Accelerometer::ParametersUpdate()
 			_priority = DEFAULT_PRIORITY;
 		}
 
+		bool calibration_changed = false;
+
 		// CAL_ACCx_OFF{X,Y,Z}
-		_offset = GetCalibrationParamsVector3f(SensorString(), "OFF", _calibration_index);
+		const Vector3f offset = GetCalibrationParamsVector3f(SensorString(), "OFF", _calibration_index);
+
+		if (Vector3f(_offset - offset).norm() > 0.001f) {
```
Comment (bkueng) [inline src/lib/sensor_calibration/Accelerometer.cpp:157]: You could avoid the square root here:
```suggestion
		if (Vector3f(_offset - offset).norm_squared() > 0.001f * 0.001f) {
```

**Review by bkueng — APPROVED** [review]

File: src/lib/sensor_calibration/Accelerometer.cpp (hunk 1 above)
Comment (dagar) [reply src/lib/sensor_calibration/Accelerometer.cpp:157]: Thanks, might as well.

**Pull request merged**